lib/Sema/TypeChecker.cpp

```diff
@@ -714,13 +714,14 @@ void swift::performTypeChecking(SourceFile &SF, TopLevelContext &TLC,
 
     typeCheckFunctionsAndExternalDecls(TC);
   }
-  MyTC.reset();
 
   // Checking that benefits from having the whole module available.
   if (!(Options & TypeCheckingFlags::DelayWholeModuleChecking)) {
     performWholeModuleTypeChecking(SF);
   }
 
+  MyTC.reset();
```

Comment:
I don't think so. performWholeModuleTypechecking() just handles some minor tasks like diagnosing @objc without import Foundation, and identifying bogus overrides. It doesn't seem like a major efficiency loss to keep the type checker alive longer.